x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/Role.java

```diff
@@ -371,4 +313,52 @@ private IndicesPermissionGroupDefinition(
             }
         }
     }
+
+    static SimpleRole buildFromRoleDescriptor(
```

Comment:
This method consolidates the specialized constructor above (`private Builder(RoleDescriptor rd, @Nullable FieldPermissionsCache fieldPermissionsCache, RestrictedIndices restrictedIndices)`)

The only production code this is used in is CompositeRoleStore, to constructor internal user roles from static role descriptors.